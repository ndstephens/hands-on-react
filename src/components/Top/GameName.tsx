import styled from '@emotion/styled';

export interface GameNameProps {
  /**
   * Text inside the header
   */
  name: string;
}

export const GameName: React.FC<GameNameProps> = ({ name }) => (
  <GameNameHeader>{name}</GameNameHeader>
);

export const GameNameHeader = styled.h1`
  font-size: 2em;
`;
