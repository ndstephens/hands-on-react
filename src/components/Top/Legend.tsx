import styled from '@emotion/styled';

export interface LegendProps {
  /**
   * Feature that should be activated after first & second actions
   */
  feature: string;
  /**
   * First action
   */
  firstAction: string;
  /**
   * Second Action
   */
  secondAction: string;
}

export const Legend: React.FC<LegendProps> = ({
  feature,
  firstAction,
  secondAction,
}) => (
  <Parent>
    <strong>{feature}: </strong>
    <Actions>
      <FirstAction>{firstAction}</FirstAction> +{' '}
      <SecondAction>{secondAction}</SecondAction>
    </Actions>
  </Parent>
);

const Parent = styled.legend`
  font-size: 1em;
  margin: 0 auto 2vw;
  line-height: 1.25em;
`;

const Actions = styled.code`
  background-color: #e3e3e3;
`;

const FirstAction = styled.span`
  color: #ec433c;
`;

const SecondAction = styled.span`
  color: #2a48ec;
`;
