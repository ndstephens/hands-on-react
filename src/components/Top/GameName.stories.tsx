import { ComponentMeta, ComponentStory } from '@storybook/react';

import { GameName } from './GameName';

export default {
  title: 'Top/GameName',
  component: GameName,
} as ComponentMeta<typeof GameName>;

const Template: ComponentStory<typeof GameName> = (args) => (
  <GameName {...args} />
);

export const MinesweeperGameName = Template.bind({});
MinesweeperGameName.args = {
  name: 'minesweeper',
};
