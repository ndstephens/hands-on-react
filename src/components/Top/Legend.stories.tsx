import { ComponentMeta, ComponentStory } from '@storybook/react';

import { Legend } from './Legend';

export default {
  title: 'Top/Legend',
  component: Legend,
} as ComponentMeta<typeof Legend>;

const Template: ComponentStory<typeof Legend> = (args) => <Legend {...args} />;

export const GameLegend = Template.bind({});
GameLegend.args = {
  feature: 'Flag',
  firstAction: 'ctrl',
  secondAction: 'click',
};
