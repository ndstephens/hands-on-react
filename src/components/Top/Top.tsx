import styled from '@emotion/styled';

import { GameName, GameNameProps } from './GameName';
import { Legend, LegendProps } from './Legend';

export const Top = ({
  children,
  ...legendProps
}: GameNameProps & LegendProps) => (
  <Header>
    <GameName>{children}</GameName>
    <Legend {...legendProps} />
  </Header>
);

const Header = styled.header`
  position: relative;
  display: inline-block;
  text-align: center;
  user-select: none;
`;
