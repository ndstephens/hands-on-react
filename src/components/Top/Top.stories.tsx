import { ComponentMeta, ComponentStory } from '@storybook/react';

import { Top } from './Top';

export default {
  title: 'Top/Top',
  component: Top,
} as ComponentMeta<typeof Top>;

const Template: ComponentStory<typeof Top> = (args) => <Top {...args} />;

export const TopPanel = Template.bind({});
TopPanel.args = {
  children: 'minesweeper',
  feature: 'Flag',
  firstAction: 'ctrl',
  secondAction: 'click',
};
