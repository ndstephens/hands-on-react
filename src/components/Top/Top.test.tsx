import { render } from '@testing-library/react';

import { Top } from './Top';

it('Top renders correctly', () => {
  const { asFragment } = render(
    <Top feature="Flag" firstAction="ctrl" secondAction="click">
      Minesweeper
    </Top>
  );

  expect(asFragment()).toMatchSnapshot();
});
