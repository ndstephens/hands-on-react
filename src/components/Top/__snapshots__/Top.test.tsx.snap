// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top renders correctly 1`] = `
<DocumentFragment>
  <header
    class="css-1bvvypo"
  >
    <h1
      class="css-18kvaxm"
    >
      Minesweeper
    </h1>
    <legend
      class="css-17qz6a2"
    >
      <strong>
        Flag: 
      </strong>
      <code
        class="css-h9xasx"
      >
        <span
          class="css-udfepp"
        >
          ctrl
        </span>
        <span
          class="css-9c23k1"
        >
          click
        </span>
      </code>
    </legend>
  </header>
</DocumentFragment>
`;
